lib/widgets/inventory/inventory_browser_app_bar.dart

```diff
@@ -241,22 +248,54 @@ class _InventoryBrowserAppBarState extends State<InventoryBrowserAppBar> {
                             if (context.mounted) {
                               ScaffoldMessenger.of(context).showSnackBar(
                                 const SnackBar(
-                                  content: Text("Selected directory is invalid"),
+                                  content:
+                                      Text("Selected directory is invalid"),
                                 ),
                               );
                             }
                             return;
                           }
                           for (var record in selectedRecords) {
-                            final uri = selectedUris == thumbUris ? record.thumbnailUri : record.assetUri;
-                            await FlutterDownloader.enqueue(
+                            final uri = selectedUris == thumbUris
+                                ? record.thumbnailUri
+                                : record.assetUri;
+                            final filename =
+                                "${record.id.split("-")[1]}-${record.formattedName.toString()}${extension(uri)}";
+                            final downloadTask = DownloadTask(
                               url: Aux.resdbToHttp(uri),
-                              savedDir: directory,
-                              showNotification: true,
-                              openFileFromNotification: false,
-                              fileName:
-                                  "${record.id.split("-")[1]}-${record.formattedName.toString()}${extension(uri)}",
+                              allowPause: true,
+                              baseDirectory: BaseDirectory.temporary,
+                              filename: filename,
+                              updates: Updates.statusAndProgress,
                             );
+                            await FileDownloader()
+                                .enqueue(downloadTask)
+                                .then((b) {
```

Comment:
Prefer putting statements on a new line instead of within a `.then()` block when the future can be awaited.

Reply:
Oh, that makes sense. It's the same in JS/TS, so I'm not sure how I managed to mess that up lol